Read me spec rev0p5

asadate opened the pull request:
Read Me changes for Spec rev0.5

rsnancollas committed:
created a support-docs directory that will contain a README and
part-spec examples. Renamed docs director to user-agreement

rsnancollas committed:
move proto versions of part specs

This commit moves proto versions of part specs to a part-spec-proto
folder. It also creates a generic component.json schema file that can be
used in conjunction with the pin definitions to define components in the
v0.3 version of the spec.

asadate committed:
Update README.md

Adding supporting documentation for the digital datasheets specifications

asadate committed:
Update README.md

rsnancollas committed:
small type fix in storage_memory.json proto file

asadate committed:
Update README.md

asadate committed:
Merge branch 'main' into ReadMe_Spec_rev0p5

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
 
 
+#### 3.7	Graph Specifications
+Datasheets include additional information through figures, graphs. Examples include but are not limited to efficiency vs load curves, power derating curves. A graph object and a curve object are specified to capture that information in the digital datasheet. The graph object focuses on defining the type of data captured, including title and axis information. The curve object focuses on capturing  ...
```
Datasheets include additional information through figures and graphs.

asadate replied:
done

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
 
 
+#### 3.7	Graph Specifications
+Datasheets include additional information through figures, graphs. Examples include but are not limited to efficiency vs load curves, power derating curves. A graph object and a curve object are specified to capture that information in the digital datasheet. The graph object focuses on defining the type of data captured, including title and axis information. The curve object focuses on capturing  ...
```
but are not limited to efficiency vs load curves and power derating curves.

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+Datasheets include additional information through figures, graphs. Examples include but are not limited to efficiency vs load curves, power derating curves. A graph object and a curve object are specified to capture that information in the digital datasheet. The graph object focuses on defining the type of data captured, including title and axis information. The curve object focuses on capturing  ...
+|Property|Description|Data Type|Required?|
+|:----|:----|:----|:----|
+|label|Description of the data in a curve|string|Yes|
```
would this need to be required if there was only one curve?

asadate replied:
Let's discuss.

asadate replied:
Removed "label" as a required property.

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|Property|Description|Data Type|Required?|
+|:----|:----|:----|:----|
+|label|Description of the data in a curve|string|Yes|
+|xData|x value of data being plotted|array|Yes|
```
array of numbers

asadate replied:
Changed

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|width|Width of the package|Number|Yes|
+|height|Height of the package|Number|Yes|
+|dimensionUnit|Unit used to describe the package dimensions|String|Yes|
+|standardPackageSize|Standard name used to designate the package size|String| |
```
what was the difference beteween standardPackageSize and standardPackageType again?

asadate replied:
0603 vs WCSP

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|length|Length of the package|Number|Yes|
+|width|Width of the package|Number|Yes|
+|height|Height of the package|Number|Yes|
+|dimensionUnit|Unit used to describe the package dimensions|String|Yes|
```
maybe mention this is basically mm or mil

asadate replied:
Added

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|standardPackageType|Standard name used to designate the package type|String| |
+
+#### 3.9	Passive Components Specifications
+The passive components digital datasheet specifications are included in GitHub: https://github.com/edatasheets/edatasheets.github.io
```
do you plan to actually like to the spec when it's up?

asadate replied:
I can put the folder location.

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+The table below give a description of the properties used to specify a resistor in a digital datasheet.
+|Property|Description|Data Type|Required?|
+|:----|:----|:----|:----|
+|Value|Resistance value|Number|Yes|
```
I'd be consistent with uppercase vs lowercase

asadate replied:
Good Catch. With JSON, these properties should all start with lower cases.

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|maxLimitingElementVoltage|Maximum voltage value that can be applied continuously to the resistor.|Number| |
+|pinSpec|Pins definition of the resistor|object| |
+|package|Package definition of the resistor|Object| |
+|minTemperature|Minimum temperature under which a resistor can be expected to reliably operate|Number| |
```
do you want the unit to be number here? I think we specify this as "SIUnit". I think I'd either clarify that this is degrees C or point to the fact that it's this more specific type we defined

(same comment applies to all of the "Number" units here

asadate replied:
Type here refers more to JSON type than Units. This is consistent with the first rev0.3 release.

asadate replied:
As discussed, I'll add JSON Data Type ( instead of Data Type) to remove any confusion

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|powerRating|Maximum power a resistor can dissipate without degrading performance|Number| |
+|temperatureCoefficient|Change in resistance when the temperature is changed|Number| |
+|maxOverloadVoltage|Maximum voltage that can be applied to the resistor for a short period of time|Number| |
+|maxLimitingElementVoltage|Maximum voltage value that can be applied continuously to the resistor.|Number| |
```
I'd remove the . here (applied continuously to the resistor) to be consistent with other lines in the table

asadate replied:
Done

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|minTemperature|Minimum temperature under which a resistor can be expected to reliably operate|Number| |
+|maxTemperature|Maximum temperature under which a resistor can be expected to reliably operate|Number| |
+|complianceList|List of compliances met by the resistor|Array| |
+|resistorDerating|Graph to capture resistor derating|Object| |
```
derating under temperature

asadate replied:
Done

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|pinSpec|Pins definition of the inductor|object| |
+|package|Package definition of the inductor|Object| |
+|complianceList|List of compliances met by the inductor component|Array| |
+|saturationCurve|Graph to capture saturation curve|Object| |
```
capture current saturation curve

asadate replied:
Done

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+#### 3.10	Power Components Specifications
+The The power component specification is included in Github at: https://github.com/edatasheets/edatasheets.github.io
+#### 3.10.1	Buck Converter Specification
+The table below gives a description of the properties used to specify a Buck (step down) switching converter in a digital datasheet.
```
technically this is actually for boosts too, right? since we specify "regulatorTopology"

asadate replied:
Yes. Most of the properties will apply to Boost converters too. The concern is that we didn't look at examples of boost converters datasheets to make sure we were capturing all the main properties. I'm ok changing the title to DC-DC converters specifications, removing references to Bucks and coming back in a later revision to make changes if we need to do so.

rsnancollas commented on support-docs/README.md:
```diff
@@ -107,7 +109,131 @@ Table below shows the list of properties used to describe pins
+|thermalShutdownThresholdRising|Thermal Shutdown Threshold with temperature rising|Number| |
+|thermalShutdownThresholdFalling|Thermal Shutdown Threshold with temperature Falling|Number| |
+|thermalShutdownHysteresis|Thermal Shutdown Hysteresis|Number| |
+|efficiency|Power efficiency of the regulator|object| |
```
maybe mention this is a graph

asadate replied:
Done

asadate committed:
Update README.md

asadate committed:
Update README.md

Added a section for the Power FET specification in the Readme file. This matches the new defined powerFETProperties object.

asadate committed:
Update component.json

asadate commented:
Need to resolve some issues with the branch
